src/main.c

```diff
@@ -1117,6 +1098,71 @@ static int get_root_dir(char *buf, size_t buf_size) {
   }
   setenv("ROOT_DIR", zl_context.root_dir, 1);
   INFO(MSG_ROOT_DIR, buf);
+
+  /* do we really need to change work dir? */
+  if (chdir(zl_context.root_dir)) {
+    DEBUG("working directory not changed - %s\n", strerror(errno));
+    return -1;
+  }
+
+  return 0;
+}
+
+static int get_workspace_dir(char *buf, size_t buf_size) {
```

Comment:
This function also seems to be overloaded:
* It gets the workspace directory
* It creates the workspace directory
* It defines stdin

Perhaps it just needs a different name? E.g. `handle_workspace_dir`, `process_workspace_dir`. Same applies to `get_root_dir`. And if you go this route, you probably don't need to pass those buffers, since we already access `zl_context` globally anyways (besides, you pass `zl_context.workspace_dir` to `get_workspace_dir` but also access it directly inside the function).